### PR TITLE
fix/85: undefined variable $_SESSION

### PR DESCRIPTION
**Pull request by TorbenBr**
| Q             | A
| ------------- | ---
| Branch        | 6.2.x
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | no
| Fixed issues  | chameleon-system/chameleon-system#85
| License       | MIT

### CODE REVIEW

File: src/CoreBundle/private/library/classes/TCMSLogChange.class.php [hunk 1]
```diff
@@ -1585,6 +1585,7 @@ public static function AddVirtualNonDbExtension($iLine, $sEntryPoint, $sClassNam
      * @param string $sClassName
      *
      * @throws ErrorException
+     * @throws DBALException
```
Comment (umulmrum) [inline src/CoreBundle/private/library/classes/TCMSLogChange.class.php]: This changes the method signature which is not allowed in a fix release. Could you wrap the exception in an ErrorException (which I normally wouldn't recommend, but here it's the only annotated exception)?

File: src/CoreBundle/private/library/classes/TCMSLogChange.class.php [hunk 2]
```diff
@@ -1597,9 +1598,10 @@ public static function deleteVirtualNonDbExtension($iLine, $sEntryPoint, $sClass
             throw new ErrorException("unable to find virtual extension {$sClassName} for {$sEntryPoint} (called from the running update in line {$iLine})", 0, E_USER_ERROR, __FILE__, __LINE__);
         }
 
-        $tableEditorManager = TTools::GetTableEditorManager($oExtension->table, $oExtension->id);
-        $tableEditorManager->AllowDeleteByAll(true);
-        $tableEditorManager->Delete($oExtension->id);
+        $connection = self::getDatabaseConnection();
+
+        $query = 'DELETE FROM '.$connection->quoteIdentifier($oExtension->table).' WHERE id = '.$connection->quote($oExtension->id);
```
Comment (umulmrum) [inline src/CoreBundle/private/library/classes/TCMSLogChange.class.php]: Could we work with a parameter for the ID? Although we need string concatenation for the table name, we should use this as little as possible.

File: src/CoreBundle/private/library/classes/TCMSLogChange.class.php (hunk 1 above)
Comment (TorbenBr) [reply src/CoreBundle/private/library/classes/TCMSLogChange.class.php]: done

**Comment by umulmrum** [comment]
Thanks @TorbenBr

**Pull request merged**